### PR TITLE
Named colors

### PR DESCRIPTION
**Pull request by IagoLast**
- Add new `color` expression allowing named colors.
- Add an example
- Add naive tests

### CODE REVIEW

File: src/core/style/expressions/color.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        const color = this._nameToRGB(this.name);
+        this.inlineMaker = () => `vec4(${(color.r / 255).toFixed(4)}, ${(color.g / 255).toFixed(4)}, ${(color.b / 255).toFixed(4)}, ${(1).toFixed(4)})`;
+    }
+
+    _nameToRGB(name) {
+        const colorRegex = /rgb\((\d{1,3}), (\d{1,3}), (\d{1,3})\)/;
+        const fakeDiv = document.createElement('div');
+        fakeDiv.style.color = name;
+        document.body.appendChild(fakeDiv);
+        const rgbSring = getComputedStyle(fakeDiv).color;
```
Comment (davidmanzanares) [inline src/core/style/expressions/color.js]: Is this browser-independent?

Comment (IagoLast) [reply src/core/style/expressions/color.js]: https://caniuse.com/#feat=getcomputedstyle

Comment (davidmanzanares) [reply src/core/style/expressions/color.js]: And that means it not only works, but that it works consistently so that the regex can work?

Comment (IagoLast) [reply src/core/style/expressions/color.js]: Good point:

I created this block:
https://bl.ocks.org/IagoLast/raw/137e831dd1f7d065a4ccd2ec2b365407/

And I'm testing against different system/browser configurations:

screen shot 2018-04-02 at 13 28 20

screen shot 2018-04-02 at 13 29 29


screen shot 2018-04-02 at 13 30 11

screen shot 2018-04-02 at 13 31 54

Comment (davidmanzanares) [reply src/core/style/expressions/color.js]: Looks good to me

File: src/core/style/expressions/color.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * const style = new carto.Style({
+ *   color: s.color('red');
+ * });
+ *
+ * @memberof carto.style.expressions
+ * @name hex
+ * @function
+ * @api
+ */
+export default class Color extends Expression {
```
Comment (davidmanzanares) [inline src/core/style/expressions/color.js]: I would change the name, the String API could be improved to automatically use these without function calls, with JS, I don't know if this is so useful, maybe `NamedColor` is better

Comment (IagoLast) [reply src/core/style/expressions/color.js]: ok, `NamedColor`

File: examples/expressions/color.html [hunk 3]
```diff
@@ -78,6 +78,10 @@
         <input type="radio" name="style" onclick="setOpacityGreen()" id="green">
         <label for="green">Opacity - Green</label>
       </li>
+      <li>
+        <input type="radio" name="style" onclick="setNamedColor()" id="named">
```
Comment (Jesus89) [inline examples/expressions/color.html]: id="orange"

File: examples/expressions/color.html [hunk 4]
```diff
@@ -78,6 +78,10 @@
...
+        <label for="green">Named Color - Orange</label>
```
Comment (Jesus89) [inline examples/expressions/color.html]: for="orange"

File: examples/expressions/color.html [hunk 5]
```diff
@@ -142,6 +149,10 @@
     function setOpacityGreen() {
       layer.blendToStyle(styleGreen);
     }
+
+    function setNamedColor(){
```
Comment (Jesus89) [inline examples/expressions/color.html]: `setNamedColorOrange`

**Comment by Jesus89** [comment]
I don't really like the syntax `namedColor('orange')`. I know we have the same issue for the column properties as in the hex colors (`hex('#ABC')`) but here there are so many letters.

Maybe @makella could help us with this, but we need to thought also about implicit cast for these text colors.

**Comment by davidmanzanares** [comment]
@Jesus89 Yes, `namedColor` is ugly, but the String API could use names without this. 

In fact, @IagoLast , it would be nice if you add that feature, it's another `else if` block in https://github.com/CartoDB/renderer-prototype/blob/master/src/core/style/parser.js#L133

Regarding the JS API, if you come up with a good name perfect, but `color` is too generic, and, in any case, I don't think this feature is very useful outside the String API scope.

**Comment by IagoLast** [comment]
@Jesus89 @davidmanzanares can you update the `named` reference in the integration tests?

**Comment by Jesus89** [comment]
Yes :point_up:

**Comment by Jesus89** [comment]
What is the final String API?

**Comment by makella** [comment]
hey guys, sorry missed this earlier. 

so what is the final notation for HEX and named colors?

**Pull request merged**